stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/querychannels/ChatEventHandler.kt

```diff
@@ -112,19 +118,23 @@ public abstract class BaseChatEventHandler : ChatEventHandler {
         }
     }
 
-    public open fun handleCidEvent(event: CidEvent, filter: FilterObject): EventHandlingResult {
+    public open fun handleCidEvent(
+        event: CidEvent,
+        filter: FilterObject,
+        cachedChannel: Channel?,
+    ): EventHandlingResult {
         return when (event) {
             is ChannelHiddenEvent -> EventHandlingResult.Remove(event.cid)
-            is MemberRemovedEvent -> handleMemberRemovedEvent(event, filter)
-            is MemberAddedEvent -> handleMemberAddedEvent(event, filter)
+            is MemberRemovedEvent -> handleMemberRemovedEvent(event, filter, cachedChannel)
+            is MemberAddedEvent -> handleMemberAddedEvent(event, filter, cachedChannel)
             else -> EventHandlingResult.Skip
         }
     }
 
-    override fun handleChatEvent(event: ChatEvent, filter: FilterObject): EventHandlingResult {
+    override fun handleChatEvent(event: ChatEvent, filter: FilterObject, cachedChannel: Channel?): EventHandlingResult {
         return when (event) {
-            is HasChannel -> handleChannelEvent(event, filter)
-            is CidEvent -> handleCidEvent(event, filter)
+            is HasChannel -> handleChannelEvent(event, filter, cachedChannel)
```

Comment:
Why do we need here `cachedChannel`? Event already provides a channel instance

Reply:
Yep, we only need the:
```
is CidEvent -> handleCidEvent(event, filter, cachedChannel)
```